pkg/kubelet/kubelet.go

```diff
@@ -1731,16 +1733,21 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(
 		if allPods.Has(string(uid)) {
 			continue
 		}
+		// If volumes have not been unmounted/detached, do not delete directory.
+		// Doing so may result in corruption of data.
 		if podVolumesExist := kl.podVolumesExist(uid); podVolumesExist {
-			// If volumes have not been unmounted/detached, do not delete directory.
-			// Doing so may result in corruption of data.
 			glog.V(3).Infof("Orphaned pod %q found, but volumes are not cleaned up; err: %v", uid, err)
 			continue
 		}
+		// Check whether volume is still mounted on disk. If so, do not delete directory
+		if volumeNames, err := kl.getPodVolumeNameListFromDisk(uid); err != nil || len(volumeNames) != 0 {
+			glog.V(3).Infof("Orphaned pod %q found, but volumes are still mounted; err: %v, volumes: %v ", uid, err, volumeNames)
```

Comment:
@luxas <https://github.com/luxas>, we do unmount unused mounted volumes,
this logic is implemented in pkg/kubelet/volumemanager/reconciler/reconciler.go
because this is the place for all mount/unmount related logic.

Please let me know if you have any questions about it.

Jing

On Tue, Aug 30, 2016 at 10:15 PM, Lucas Käldström <notifications@github.com>
wrote:

> In pkg/kubelet/kubelet.go
> <https://github.com/kubernetes/kubernetes/pull/27970#discussion_r76925660>
> :
>
> >  			glog.V(3).Infof("Orphaned pod %q found, but volumes are not cleaned up; err: %v", uid, err)
> >  			continue
> >  		}
> > +		// Check whether volume is still mounted on disk. If so, do not delete directory
> > +		if volumeNames, err := kl.getPodVolumeNameListFromDisk(uid); err != nil || len(volumeNames) != 0 {
> > +			glog.V(3).Infof("Orphaned pod %q found, but volumes are still mounted; err: %v, volumes: %v ", uid, err, volumeNames)
>
> Why not unmount unused mounted volumes here?
> I think that's what #31596
> <https://github.com/kubernetes/kubernetes/issues/31596> is asking for
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/27970/files/f19a1148db1b7584be6b6b60abaf8c0bd1503ed3#r76925660>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASSNxZ2PPiHykY7mWc1u4e7Uagxqpgsgks5qlQ3WgaJpZM4I9M0t>
> .
>



-- 
- Jing